### ISSUE TITLE
Performance issue in API aggregation layer

### ISSUE THREAD

**Issue by qiujian16** [kubernetes/kubernetes#74280]
**What happened**:
I have two APIServices installed on api aggregator. When I have a lot of clients (more than 200) calling these two APIs at the same time via kube-apiserver, I see a lot of timeouts in client, and when i run kubectl to get these api resource, the response time becomes as much as more than 10s.

It seems that kube-apiserver will open a socket for each APIService, since I saw to established connections to the aggregate apiserver:
```
[root@mcm01-master ~]# netstat -plan | grep 10.1.131.196
tcp        0      0 10.1.223.64:33292       10.1.131.196:443        ESTABLISHED 17402/hyperkube
tcp        0      0 10.1.223.64:33286       10.1.131.196:443        ESTABLISHED 17402/hyperkube
```
However, when there are too many requests, one connection may not be able to handle all of them?

To work around, I enable "--enable-aggregator-routing=true" on kube-apiserver, and increase the replica of aggregate apiserver. kube-apiserver will have 2 connections for each of the aggregate-apiserver pod, and it currently helps to resolve my problem.

**What you expected to happen**:
kube-apiserver should be able to handle large amount of api requests served by aggregate apiserver.

**How to reproduce it (as minimally and precisely as possible)**:
Start a aggregate apiserver, start a lot of clients to call API through kube-apiserver

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.12.4
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): centos
- Kernel (e.g. `uname -a`):  3.10.0-862.14.4.el7.x86_64
- Install tools:
- Others:

**Comment by liggitt** [comment]
might be related to https://github.com/golang/go/issues/27044, if aggregator<->backend is using http/2 connections

**Comment by qiujian16** [comment]
thanks @liggitt seems it is fixed in golang 1.12, i will try kube-apiserver with golang 1.12 to verify.

**Comment by qiujian16** [comment]
thanks @liggitt, seems it will be fixed in golang 1.12...

**Comment by liggitt** [comment]
in 1.14, upgraded to go1.12, multiple backend connections are opened if required

/close
